spec/posthog/feature_flag_spec.rb

```diff
@@ -3322,6 +3324,346 @@ class PostHog
+            "filters": {
+                "groups": [
+                    {
+                        "properties": [
+                            {
+                                "key": "region",
+                                "operator": "exact",
+                                "value": ["USA"],
+                                "type": "person",
+                            }
+                        ],
+                        "rollout_percentage": 100,
+                    }
+                ],
+                "payloads": {"true": 300},
+            },
+      }
+      stub_request(
+        :get,
+        'https://app.posthog.com/api/feature_flag/local_evaluation?token=testsecret'
+      ).to_return(status: 200, body: {"flags": [basic_flag_res]}.to_json)
+
+      stub_request(:post, decide_endpoint)
+      .to_return(status: 200, body:{"featureFlags": {"person-flag" => "person-flag-value"}}.to_json)
+
+      c = Client.new(api_key: API_KEY, personal_api_key: API_KEY, test_mode: true)
+      expect(c.get_feature_flag_payload("person-flag", "some-distinct-id", person_properties: {"region" => "USA"})).to eq(300)
+      expect(c.get_feature_flag_payload("person-flag", "some-distinct-id", match_value: true, person_properties: {"region" => "USA"})).to eq(300)
+      assert_not_requested :post, decide_endpoint
+    end
+
+    it 'evaluates boolean feature flags with decide' do
+      basic_flag_res = {
+        "id": 1,
+            "name": "Beta Feature",
+            "key": "person-flag",
+            "is_simple_flag": true,
+            "active": true,
+            "filters": {
+                "groups": [
+                    {
+                        "properties": [
+                            {
+                                "key": "region",
+                                "operator": "exact",
+                                "value": ["USA"],
+                                "type": "person",
+                            }
+                        ],
+                        "rollout_percentage": 100,
+                    }
+                ],
+                "payloads": {"true": 300},
+            },
+      }
+      stub_request(
+        :get,
+        'https://app.posthog.com/api/feature_flag/local_evaluation?token=testsecret'
+      ).to_return(status: 200, body: {"flags": [basic_flag_res]}.to_json)
+      stub_request(:post, decide_endpoint)
+        .to_return(status: 200, body:{"featureFlagPayloads": {"person-flag" => 300}}.to_json)
+      c = Client.new(api_key: API_KEY, personal_api_key: API_KEY, test_mode: true)
+      expect(c.get_feature_flag_payload("person-flag", "some-distinct-id", person_properties: {"region": "USA"})).to eq(300)
+      expect(c.get_feature_flag_payload("person-flag", "some-distinct-id", match_value: true, person_properties: {"region": "USA"})).to eq(300)
+      # assert_requested :post, decide_endpoint, times: 2
+    end
+
+    it 'with multivariate feature flag payloads' do
```

Comment:
we need a test with a json payload coming from decide, to ensure it's parsed properly

Reply:
@neilkakkar ^ ?

Reply:
Ah, I meant that all payloads are string/int, but I don't see an array or json payload.

Just updating that test above to have real json payloads would work :)